setup.py

```diff
@@ -64,7 +64,12 @@
     packages=find_packages(exclude=("tests", "docs")),
     python_requires=">=3.6",
     py_modules=["vyper"],
-    install_requires=["asttokens==2.0.4", "pycryptodome>=3.5.1,<4", "semantic-version==2.8.5"],
+    install_requires=[
+        "asttokens==2.0.4",
+        "pycryptodome>=3.5.1,<4",
+        "semantic-version==2.8.5",
+        "cached-property==1.5.2",
```

Comment:
You can import this backport conditionally for Python <3.8
```suggestion
        "cached-property==1.5.2 ; python_version<'3.8'",
```